### PR TITLE
Updated to use more recent dependencies

### PR DESCRIPTION
**Pull request by miballe**
- When cloning the environment it seems that some dependencies generate warning since are quite old. Changed to more recent versions and changed the version to >= instead of ==.
- Updated the docker file to use a Python 3 image.
- Removed the .python file that redirected the interpreter in some environments
- Minor updates in the documentation aligned to above changes
- Version update to avoid cache issues, and because makes sense after few non-breaking changes were applied.
- Python 2 is being deprecated on the 1st of January 2020. See https://www.python.org/doc/sunset-python-2/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -76,4 +76,8 @@ the subscription *and* consuming it. Instead they will just consume an already e
 - [changed] - Removed maximum messages parameter in listen call. It was unnecessary.
 - [changed] - Added test shell script './dnaStreaming/test/test_run_docker.sh' for testing Docker and streams; to be used only with DNA Engineering assistance.
   
-
+1.4.2 / 2018-03-19
```
Comment (madmanu) [inline CHANGELOG.md]: You may have forgotten to change the date :)

Comment (miballe) [reply CHANGELOG.md]: Fixed!

**Comment by charleyferrari** [comment]
@miballe Looks like there are issues with pubsub 2.0. I got the listener working by using your requirements and changing the pubsub version to `google-cloud-pubsub==1.7.0`

**Comment by zakagan** [comment]
I believe this PR is outdated, so it will be shortly closed

**Pull request closed without merging**